src/tokenizer/types.js

```diff
@@ -80,7 +80,7 @@ export const types = {
   doubleColon: new TokenType("::", { beforeExpr }),
   dot: new TokenType("."),
   question: new TokenType("?", { beforeExpr }),
-  arrow: new TokenType("=>", { beforeExpr }),
+  arrow: new TokenType("=>", { beforeExpr, startsExpr }),
```

Comment:
ideally this would only happen when `this.hasPlugin("lightscript")`, but I don't think there's an easy way to do that yet. So this is fine for now.